src/extra/JsonSchemaRenderer/index.tsx

```diff
@@ -0,0 +1,213 @@
+import * as React from 'react';
+import { forEach, keys, pick, get, isEqual, isArray } from 'lodash';
+import jsone from 'json-e';
+import ajv from 'ajv';
+import asRendition from '../../asRendition';
+import {
+	DefaultProps,
+	RenditionSystemProps,
+	Theme,
+	Tooltip,
+} from '../../common-types';
+import { Flex } from '../../components/Flex';
+import Alert from '../../components/Alert';
+import widgets, { WidgetWrapperUiOptions, WidgetMeta } from './widgets';
+import { Value, JSONSchema, UiSchema, Format } from './types';
+
+interface WrapperProp extends Tooltip {
+	children: React.ReactNode;
+}
+
+const Wrapper = ({ tooltip, children, ...props }: WrapperProp) => (
+	<Flex
+		tooltip={tooltip}
+		minWidth={0}
+		minHeight={0}
+		alignItems="flex-start"
+		mb={1}
+		flexDirection="column"
+		{...props}
+	>
+		{children}
+	</Flex>
+);
+
+export const getValue = (value?: Value, uiSchema?: UiSchema) => {
+	return get(uiSchema, 'ui:value', value);
+};
+
+export const getType = (value?: Value) => {
+	if (value === undefined) {
+		return 'default';
+	}
+	if (value === null) {
+		return 'null';
+	}
+	return isArray(value) ? 'array' : typeof value;
+};
+
+export const getWidget = (value?: Value, uiSchemaWidget?: string) => {
+	const typeWidgets = get(widgets, getType(value), widgets.default);
+	return get(typeWidgets, uiSchemaWidget || 'default', widgets.default.default);
+};
+
+const processValue = ({
+	value,
+	uiSchema,
+}: Pick<InternalJsonSchemaRendererProps, 'value' | 'uiSchema'>) => {
+	return getValue(value, uiSchema);
+};
+
+class JsonSchemaRenderer extends React.PureComponent<
```

Comment:
Any reason not to implement this with hooks, and just wrap the export in an Error boundary?

Reply:
Ok - I've created an `ErrorBoundary` component in the `internal` directory and wrapped the new functional `JsonSchemaRenderer` component with it. Let me know if that's what you're thinking or if there's a cleaner way.